src/runtime_src/core/common/ishim.h

```diff
@@ -431,15 +431,30 @@ struct shim : public DeviceType
   void
   p2p_enable(bool force) override
   {
-    if (auto ret = xclP2pEnable(DeviceType::get_device_handle(), true, force))
-      throw system_error(ret, "failed to enable p2p");
+    auto ret = xclP2pEnable(DeviceType::get_device_handle(), true, force);
+    switch (ret) {
```

Comment:
These two values stem from shim.cpp:1134 and shim.cpp:2662. This matches processing for set_cma (ishim.h:461) but it is mixed with some exception handling returns which makes the default case necessary. I will be sure to add a few comments if you like how this is formatted